packages/core-web/src/services/LightningService.ts

```diff
@@ -47,6 +47,66 @@ export class LightningService {
     })
   }
 
+  async createInvoiceTweakedWithGateway(
+    amount: MSats,
+    description: string,
+    expiryTime: number | null = null, // in seconds
+    index: number,
+    extraMeta: JSONObject = {},
+    gatewayInfo: GatewayInfo,
+  ): Promise<CreateBolt11Response> {
+    return await this.client.rpcSingle(
+      'ln',
+      'create_bolt11_invoice_for_user_tweaked',
+      {
+        amount,
+        description,
+        expiry_time: expiryTime,
+        index,
+        extra_meta: extraMeta,
+        gateway: gatewayInfo,
+      },
+    )
+  }
+
+  async createInvoiceTweaked(
+    amount: MSats,
+    description: string,
+    expiryTime: number | null = null, // in seconds
+    publicKey: string,
+    index: number,
+    extraMeta: JSONObject = {},
+  ): Promise<CreateBolt11Response> {
+    await this.updateGatewayCache()
+    const gateway = await this._getDefaultGatewayInfo()
+    return await this.client.rpcSingle(
+      'ln',
+      'create_bolt11_invoice_for_user_tweaked',
+      {
+        amount,
+        description,
+        expiry_time: expiryTime,
+        user_key: publicKey,
+        index,
+        extra_meta: extraMeta,
+        gateway: gateway.info,
+      },
+    )
+  }
+
+  // Returns the operation ids of payments received to the tweaks of the user secret key
+  async scanReceivesForTweaks(
+    userKey: string,
```

Comment:
is this just a pubkey? 

It doesn't look like the user provides a key in the `createInvoiceTweaked` function, so how does the user get this key?